nameresolution/mdns/mdns.go

```diff
@@ -300,9 +292,35 @@ func (m *Resolver) Init(metadata nameresolution.Metadata) error {
 	}
 
 	m.logger.Infof("local service entry announced: %s -> %s:%d", appID, hostAddress, port)
+
+	resolver, err := m.getZeroconfResolver()
+	if err != nil {
+		return err
+	}
+	m.resolver.Store(resolver)
+
 	return nil
 }
 
+func (m *Resolver) getZeroconfResolver() (resolver *zeroconf.Resolver, err error) {
+	// Try with IPv4 + IPv6 first, then IPv4-only, then IPv6-only
+	opts := []zeroconf.ClientOption{
+		zeroconf.SelectIPTraffic(zeroconf.IPv4AndIPv6),
+		zeroconf.SelectIPTraffic(zeroconf.IPv4),
+		zeroconf.SelectIPTraffic(zeroconf.IPv6),
+	}
+	for i := 0; i < len(opts); i++ {
+		resolver, err = zeroconf.NewResolver(opts[i])
+		if err == nil {
+			break
+		}
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize resolver: %w", err)
```

Comment:
This will only return the ipv6 specific error which could mask the fact that ipv4 is also broken. Can you update the error message to make this clearer?